TestHelper.psm1

```diff
@@ -201,3 +204,210 @@ function Install-ResourceDesigner {
+        The name of the DSC Module containing the resource that the tests will be
+        run on.
+
+    .PARAMETER DSCResourceName
+        The full name of the DSC resource that the tests will be run on. This is
+        usually the name of the folder containing the actual resource MOF file.
+
+    .PARAMETER TestType
+        Specifies the type of tests that are being intialized. It can be:
+        Unit: Initialize for running Unit tests on a DSC resource. Default.
+        Integration: Initialize for running Integration tests on a DSC resource.
+
+    .OUTPUT
+        Returns a test environment object which must be passed to the
+        Restore-TestEnvironment function to allow it to restore the system
+        back to the original state as well as clean up and working/temp files.
+        
+    .EXAMPLE
+        $TestEnvironment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            
+        This command will initialize the test enviroment for Unit testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.      
+
+    .EXAMPLE
+        $TestEnvironment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            -TestType Integration
+            
+        This command will initialize the test enviroment for Integration testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.    
+#>
+function Initialize-TestEnvironment
+{
+    [OutputType([PSObject])]
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCModuleName,
+        
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCResourceName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('Unit','Integration')]
+        [String] $TestType
+    )
+    
+    Write-Verbose -Verbose (`
+        'Initializing Test Environment for {0} testing of {1} in module {2}.' `
+            -f $TestType,$DSCResourceName,$DSCModuleName)   
+    if ($TestType -eq 'Unit')
+    {
+        [String] $RelativeModulePath = "DSCResources\$DSCResourceName\$DSCResourceName.psm1"
+    }
+    else
+    {
+        [String] $RelativeModulePath = "$DSCModuleName.psd1"
+    }
+
+    # Unique Temp Working Folder - always gets removed on completion
+    # The tests can put anything in here and it will get cleaned up.
+    [String] $RandomFileName = [System.IO.Path]::GetRandomFileName()
+    [String] $WorkingFolder = Join-Path -Path $env:Temp -ChildPath "$DSCResourceName_$RandomFileName" 
+    # Create the working folder if it doesn't exist (it really shouldn't anyway)
+    if (-not (Test-Path -Path $WorkingFolder))
+    {
+        New-Item -Path $WorkingFolder -ItemType Directory
+    } 
+    
+    # The folder where this module is found
+    [String] $moduleRoot = $pwd
```

Comment:
Was testing it on my machine and looks like due to this line we have an assumption that we will run those functions from the root folder (e.g. C:\resources\xTimeZone). 
I think it's a bit limiting since users may want to run just subset of tests while invoking them from e.g. C:\resources\xTimeZone\tests\Unit etc.

I'm not a big fan of $pwd since it depends on location from which script is called, I usually use:
```
$rootPath = Split-Path -Parent $MyInvocation.MyCommand.Path
```